macros/gemPlotterAllChannels.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# Usage:
+# ./gemPlotterAllChannels.sh <InFile> <anaType> <branchName>
+
+INFILE=$1
+ANATYPE=$2
+BRANCHNAME=$3
+
+for chan in {0..127}
+do
+    if [ $chan -eq 0 ]; then
+        echo gemPlotter.py -i$INFILE -s$chan -a --anaType=$ANATYPE --branchName=$BRANCHNAME --rootOpt=RECREATE
+        gemPlotter.py -i$INFILE -s$chan -a --anaType=$ANATYPE --branchName=$BRANCHNAME --rootOpt=RECREATE
+    else
+        echo gemPlotter.py -i$INFILE -s$chan -a --anaType=$ANATYPE --branchName=$BRANCHNAME --rootOpt=UPDATE
+        gemPlotter.py -i$INFILE -s$chan -a --anaType=$ANATYPE --branchName=$BRANCHNAME --rootOpt=UPDATE
+    fi
```

Comment:
How does this behave if the user doesn't follow the syntax?
Are there "safe" defaults?
If not, you should add a `usage` function that checks the arguments, I should have a few examples laying around

Reply:
No if the user doesn't follow the syntax all hell breaks loose (read "undefined behavior could occur). 

Could you point me to one of your examples?  I can then use that to make a `usage` function.

Reply:
Something simple like (after your shebang):
```bash
usage() {
    echo 'Usage : ./gemPlotterAllChannels.sh <InFile> <anaType> <branchName>'
    echo '         InFile: description'
    echo '         anaType: description'
    echo '         branchName: description'
    kill -INT $$;
}

## could also add some option parsing with getopts or similar, but probably not necessary
inf=${1}
ana=${2}
branch=${3}
## then here you check the variables that are necessary:
if [ -z ${inf+x} ] || [ -z ${ana+x} ] || [ -z ${branch+x} ]
# or, without the assignment above
if [ -z ${3+x} ] || [ -z ${2+x} ] || [ -z ${3+x} ]
then
    usage
fi
```